net/src/ipc_net_worker.rs

```diff
@@ -119,13 +149,91 @@ impl IpcNetWorker {
                 let out: Box<NetWorker> = Box::new(IpcClient::new(h, socket, block_connect)?);
                 Ok(out)
             }),
+            None,
         )
     }
 
     // -- private -- //
 
+    /// create a new IpcNetWorker instance pointing to a process that we spawn here
+    fn priv_spawn(
+        handler: NetHandler,
+        cmd: String,
+        args: Vec<String>,
+        work_dir: String,
+        env: HashMap<String, String>,
+        block_connect: bool,
+    ) -> NetResult<Self> {
+        let mut proc = std::process::Command::new(cmd);
+        proc.stdout(std::process::Stdio::piped());
+        proc.args(&args);
+        proc.envs(&env);
+        proc.current_dir(work_dir);
+        let mut proc = proc.spawn()?;
+
+        let re_ready = regex::Regex::new("#IPC-READY#")?;
+        let re_ipc = regex::Regex::new("(?m)^#IPC-BINDING#:(.+)$")?;
+        let re_p2p = regex::Regex::new("(?m)^#P2P-BINDING#:(.+)$")?;
+
+        let mut ipc_binding = "".to_string();
+        let mut p2p_bindings: Vec<String> = Vec::new();
+
+        if let Some(ref mut stdout) = proc.stdout {
```

Comment:
So... I don't really get what the purpose of this whole codeblock is. It seems like you are reading the output of the subprocess. Could you elaborate?

Reply:
good call. added some comments.

You are correct, we need to know the child process is up and running before we try to connect to it over ipc. It monitors stdout, gleans some binding info, and then proceeds when it gets the `#IPC-READY#` signal.